plugins/file-open-save-new/api/src/main/java/org/pentaho/di/plugins/fileopensave/api/providers/Entity.java

```diff
@@ -37,4 +39,14 @@ public interface Entity {
   boolean isCanEdit();
   boolean isCanDelete();
   EntityType getEntityType();
+
+  /**
+   * Gets the decoded version of name of an element.
+   * Defaults to getName() {@code String}.
+   *
+   * @return A non-empty name of an element.
+   */
+  default String getNameDecoded() throws FileSystemException {
```

Comment:
We should try to keep Apache VFS types as an implementation detail.
You can use the existing exception, `org.pentaho.di.plugins.fileopensave.api.providers.exception.FileException` instead.

Reply:
Addressed